src/Language/PureScript/Errors.hs

```diff
@@ -612,13 +612,20 @@ prettyPrintSingleError (PPEOptions codeColor full level showDocs) e = flip evalS
                                                           _
                                                           (Just (PartialConstraintData bs b)))) =
       paras [ line "A case expression could not be determined to cover all inputs."
-            , line "The following additional cases are required to cover all inputs:\n"
+            , line "The following additional cases are required to cover all inputs:"
             , indent $ paras $
                 Box.hsep 1 Box.left
                   (map (paras . map (line . markCode)) (transpose bs))
                   : [line "..." | not b]
             , line "Alternatively, add a Partial constraint to the type of the enclosing value."
             ]
+    renderSimpleErrorMessage (NoInstanceFound (Constraint C.Discard [ty] _)) =
+      paras [ line "A result of type"
+            , markCodeBox $ indent $ typeAsBox ty
+            , line "was implicitly discarded in a do notation block."
+            , line ("You can use " <> markCode "_ <- ..." <> " to explicitly discard the result.")
+            , line "Alternatively, add a Discard constraint to the type of the discarded value."
```

Comment:
I was thinking maybe we'd need to add a `Discard` constraint if the type were polymorphic, but actually that seems very unlikely.